### PR TITLE
DO NOT MERGE: fix #1026

### LINKED ISSUE #1026: Allow (pre)processing multiple times
If you have preprocessed a prep template or processed a preprocessed data, you cannot re-(pre)process them. This is a limitation on the GUI, not in the database.

### PR DESCRIPTION
**Pull request by antgonza**
This allows the user to process the same data multiple times. Note that: 
(1) This allows to pick against the same reference multiple times, not sure if we should prevent this.
(2) Added the processing info into the preprocessing page. Not really happy with the display so open to suggestions.

### CODE REVIEW

File: qiita_pet/uimodules/preprocessed_data_tab.py [hunk 1]
```diff
@@ -40,7 +42,11 @@ def render(self, study_id, preprocessed_data):
         is_local_request = self._is_local()
         show_ebi_btn = user.level == "admin"
         processing_status = preprocessed_data.processing_status
-        processed_data = preprocessed_data.processed_data
+        processed_data = [(id_,
```
Comment (wasade) [inline qiita_pet/uimodules/preprocessed_data_tab.py]: this is pretty confusing to read, possible to break it up?

File: qiita_pet/uimodules/preprocessed_data_tab.py [hunk 2]
```diff
@@ -40,7 +42,11 @@ def render(self, study_id, preprocessed_data):
...
+                           '<br/>'.join(['<b>%s:</b> %s' % (k, v) for k, v in
```
Comment (wasade) [inline qiita_pet/uimodules/preprocessed_data_tab.py]: minor, but the formatting will implicitly split up the tuple so the expansion of `(k, v)` isn't necessary

File: qiita_pet/uimodules/preprocessed_data_tab.py (hunk 1 above)
Comment (antgonza) [inline qiita_pet/uimodules/preprocessed_data_tab.py]: I can, do you have suggestions? I decided to left it like this to avoid multiple separate loops ...

Comment (wasade) [inline qiita_pet/uimodules/preprocessed_data_tab.py]: hows something like this?

```python
infos = []
fmt = '<b>%s:</b %s'
for id_ in preprocessed_data.processed_data:
    info = ProcessedData(id_).processing_info
    infos.append((id_, '<br/>'.join(fmt % kv for kv in view_items(info)))
```

**Comment by josenavas** [comment]
Thanks for working on this @antgonza 

I don't agree on having the processing information in there. Note that such information belongs to the processed data, not to the preprocessed data, so it is confusing having it in there. The idea of having the processed data id to be a link was to direct the user to the tab so they can check that information in there.

I would suggest moving that information to the processed data tab, and leave the preprocessed data tab just with a list of links. Note that this is the approach used in the prep template.

Also, how are you storing the processing status? If I understand correctly, the database only allows to store 1 processing status at a time. The user can potentially start 2 different processings (let's say against GG and against Silva) but only one status will be stored. Furthermore, if they fail, the user will be only able to see one of the errors. I think this requires some changes to the DB, so this issue is bigger that we thought at the beginning.

Re your first point: I would suggest that we need to prevent processing two times using the exact same parameters set, as it won't change the results, so we are wasting compute time. However, if the user tries to process using different parameters agains the same reference, then we should allow this (I think right now we have only one parameter set, but see #1161)

**Pull request closed without merging**